### PR TITLE
feat: #2537 Improve UI for no-result of search's feature in "Developer Development" page 

### PR DESCRIPTION
**Pull request by ant066**
# Pull request checklist

## Does this close any currently open issues?

fixes: #2537

**Please check if your PR fulfills the following requirements:**

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures
- [x] Test (`yarn test`) has passed locally and any fixes were made for failures

## Pull request type

**Please check the type of change your PR introduces:**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Screen Shot 2020-09-03 at 10 48 44 AM
Issue Number: #2537

## What is the new behavior?

Screen Shot 2020-09-03 at 10 48 31 AM

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/admin-portal/src/components/pages/devs-management/devs-management.tsx [hunk 1]
```diff
@@ -181,49 +180,38 @@ export const DevsManagement: React.FC = () => {
...
-        <>
-          <Section>
-            <Table scrollable={true} loading={false} data={data || []} columns={columns} />
-          </Section>
-          <Section>
-            <div>Total: {totalCount}</div>
-          </Section>
-          <Pagination onChange={onPageChange} totalCount={totalCount} pageSize={pageSize} pageNumber={pageNumber} />
-          {/* <SetDeveloperStatusModal
+      {isLoading || !data ? <Loader /> : renderResult(data, columns, totalCount)}
```
Comment (vuhuucuong) [inline packages/admin-portal/src/components/pages/devs-management/devs-management.tsx:189]: I think you can remove `isLoading || !data` and move it to renderResult`:

`renderResult(isLoading, data, columns, totalCount)`
and then in `renderResult`
`if (isLoading || !data) {return <Loader />}`
`if (data?.length === 0){
	return <Alert message="No Results " type="info" />
}`

It's better to move logic into one place

**Review by vuhuucuong — APPROVED** [review]
Other LGTM

File: packages/admin-portal/src/components/pages/devs-management/devs-management.tsx [hunk 2]
```diff
@@ -181,49 +180,38 @@ export const DevsManagement: React.FC = () => {
...
-        <Link className="text-center" to={Routes.DEV_MANAGEMENT}>
-          <Button variant="primary" type="button">
-            New Search
-          </Button>
-        </Link>
-      </React.Fragment>
-    )
-  }
-
   return (
```
Comment (tanphamhaiduong) [inline packages/admin-portal/src/components/pages/devs-management/devs-management.tsx:183]: I think
```
if (isLoading) {
  return <Loader />
}
```

File: packages/admin-portal/src/components/pages/devs-management/devs-management.tsx (hunk 1 above)
Comment (tanphamhaiduong) [reply packages/admin-portal/src/components/pages/devs-management/devs-management.tsx:189]: you can remove isLoading here, just return Loader before return component

File: packages/admin-portal/src/components/pages/devs-management/devs-management.tsx [hunk 3]
```diff
@@ -181,49 +180,38 @@ export const DevsManagement: React.FC = () => {
...
     </ErrorBoundary>
   )
 }
 
+export const renderResult = (data, columns, totalCount) => {
+  return data?.length === 0 ? (
+    <Alert message="No Results " type="info" />
+  ) : (
+    <>
+      <Section>
```
Comment (tanphamhaiduong) [inline packages/admin-portal/src/components/pages/devs-management/devs-management.tsx:207]: If you move Total to render content, pls check consistent between dev and apps management

File: packages/admin-portal/src/components/pages/devs-management/devs-management.tsx (hunk 2 above)
Comment (ant066) [reply packages/admin-portal/src/components/pages/devs-management/devs-management.tsx:183]: if we render the `Loader` here, the page will rerender the filter form when fetching data, I think it not necessary. I think I move the Loader to `renderResult` function for more readable.

File: packages/admin-portal/src/components/pages/devs-management/devs-management.tsx (hunk 3 above)
Comment (ant066) [reply packages/admin-portal/src/components/pages/devs-management/devs-management.tsx:207]: Sure, the apps management currently renders like below. I move the `total` to the top of the result to make it more consistent.
Screen Shot 2020-09-03 at 2 44 55 PM

**Review by tanphamhaiduong — APPROVED** [review]

**Pull request merged**